server/game/cards/13.6-LMHR/RecruitFromTheDungeons.js

```diff
@@ -0,0 +1,28 @@
+const DrawCard = require('../../drawcard.js');
+
+class RecruitFromTheDungeons extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Have character participate in the challenge',
+            condition: context => this.game.isDuringChallenge({ defendingPlayer: context.player }) && this.canParticipateInChallenge(),
+            cost: ability.costs.kneelSelf(),
+            handler: context => {
+                this.challengeToParticipateIn = this.game.currentChallenge;
+                this.challengeToParticipateIn.addDefender(this);
+                this.game.addMessage('{0} kneels {1} to have {1} participate in the challenge on their side', context.player, this);
+                this.game.once('afterChallenge', this.afterChallenge.bind(this.challengeToParticipateIn, context));

```

Comment:
`this.game.once` is the correct method here. It's the makeshift way we currently do delayed effects. It will set up an event listener for the `afterChallenge` event that will be called the next time that event is emitted, then stop listening. It works well for `afterChallenge` in particular because that event is raised only once per challenge.

The second parameter is not correct currently. `this.afterChallenge.bind(this.challengeToParticipateIn, context)` is difficult to explain the space of a comment, but basically it will attempt to call the `afterChallenge` method defined on the current class using the challenge object (`this.challengeToParticipateIn`) as the `this` object inside that method. Which will crash as it tries to move `this` (which is now the challenge, not the card) to shadows.

The correct way to use bind in this instance would be:
```javascript
this.game.once('afterChallenge', this.afterChallenge.bind(this, this.challengeToParticipateIn, context));
```

However, it would be much simpler to write (and read later as):
```javascript
this.game.once('afterChallenge', () => this.afterChallenge(this.challengeToParticipateIn, context));
```

Reply:
thank you for the feedback!